compiler/rustc_typeck/src/structured_errors/wrong_number_of_generic_args.rs

```diff
@@ -296,12 +296,11 @@ impl<'a, 'tcx> WrongNumberOfGenericArgs<'a, 'tcx> {
         let fn_sig = self.tcx.hir().get_if_local(self.def_id).and_then(fn_sig);
         let is_used_in_input = |def_id| {
             fn_sig.map_or(false, |fn_sig| {
-                fn_sig.decl.inputs.iter().any(|ty| match ty.kind {
-                    hir::TyKind::Path(hir::QPath::Resolved(
-                        None,
-                        hir::Path { res: hir::def::Res::Def(_, id), .. },
-                    )) if *id == def_id => true,
-                    _ => false,
+                fn_sig.decl.inputs.iter().any(|ty| {
+                    matches!(ty.kind, hir::TyKind::Path(hir::QPath::Resolved(
+                 None,
+                 hir::Path { res: hir::def::Res::Def(_, id), .. },
+                  )) if *id == def_id)
```

Comment:
In general, should we avoid using `matches!` with arm guards? Can't we just simplify the original match by moving the guard into the actual arm?

Reply:
LIke this, right?

````rust
        let is_used_in_input = |def_id| {
            fn_sig.map_or(false, |fn_sig| {
                fn_sig.decl.inputs.iter().any(|ty| match ty.kind {
                    hir::TyKind::Path(hir::QPath::Resolved(
                        None,
                        hir::Path { res: hir::def::Res::Def(_, id), .. },
                    )) => *id == def_id,
                    _ => false,
                })
            })
        };
````